server/apps/archive/common.py

```diff
@@ -169,16 +169,9 @@ def generate_unique_id_and_name(item, repo_type=ARCHIVE):
     """
 
     try:
-        key_name = 'TEST_{}_SEQ'.format(repo_type.upper()) if superdesk.app.config.get('SUPERDESK_TESTING', False) \
-            else '{}_SEQ'.format(repo_type.upper())
-
-        unique_id = update_key(key_name, flag=True)
-
-        if unique_id:
-            item['unique_id'] = unique_id
-            item['unique_name'] = "#" + str(unique_id)
-        else:
-            raise IdentifierGenerationError()
+        unique_id = str(uuid1())
```

Comment:
As per requirement we need both. `unique_id` is for internal storage and is not editable. But `unique_name` can be changed by user and also user can open a story using `unique_name (ctrl+0) is shortcut key`. 

More information can be found in https://wiki.sourcefabric.org/display/NR/Content+Metadata